packages/react-native/ReactAndroid/src/main/java/com/facebook/react/internal/views/text/EffectiveTextAttributeProvider.kt

```diff
@@ -0,0 +1,24 @@
+package com.facebook.react.internal.views.text
+
+import com.facebook.react.common.assets.ReactFontManager
+import com.facebook.react.views.text.TextTransform
+
+/**
+ * Interface for an entity providing effective text attributes of a text node/fragment
+ */
+interface EffectiveTextAttributeProvider : BasicTextAttributeProvider {
+  companion object {
+    const val UNSET = ReactFontManager.TypefaceStyle.UNSET
+  }
```

Comment:
Do we need this redefined here? Why?

Reply:
Yeah, it's smelly... `TypefaceStyle` is a tuple in the form `TypefaceStyle(int weight, boolean italic)`, while I'm using this for font size. So not completely the same thing. So I kind of "stole" their constant...
 
Possible ways forward here:

- Switching to `Int?` for `effectiveFontSize` (with some possible minor performance penalty)
- Redeclaring `UNSET = -1` in this interface
- Refactoring these two constants into something like `FontConstants.UNSET`

Reply:
I'd prefer the `FontConstants.UNSET` path, as it doesn't introduce inconsistency with the existing code and feels acceptably clean.

Reply:
You can quickly search for `UNSET =` in the repository. In a few places, we define it on a per-class basis. It's typically `-1`, but also a minimal `Long` value in one place.

Reply:
The issue starts here:

https://github.com/facebook/react-native/blob/285629922cafd86914af9ed95ff1fbbd2c5a7c79/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/text/ReactTextShadowNode.java#L83

`ReactTextShadowNode` uses the constant from `ReactBaseTextShadowNode`...

https://github.com/facebook/react-native/blob/285629922cafd86914af9ed95ff1fbbd2c5a7c79/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/text/ReactBaseTextShadowNode.java#L58

...which in turn "steals" it from `TypefaceStyle`.

So this is what I originally replicated and you asked about.

Reply:
So, actually, `FontConstants` is also not a good name/source for this constant. In my latest revision, I suggest an approach where all files I create/modify depend on the `TextAttributeProps.UNSET` static property.

This reduces the total number of `UNSET = -1` definitions in the repository to three.